pylint/lint/pylinter.py

```diff
@@ -1014,7 +1020,24 @@ def _check_files(self, get_ast, file_descrs):
         """
         with self._astroid_module_checker() as check_astroid_module:
             for name, filepath, modname in file_descrs:
-                self._check_file(get_ast, check_astroid_module, name, filepath, modname)
+                error = None
+                try:
+                    self._check_file(
+                        get_ast, check_astroid_module, name, filepath, modname
+                    )
+                except Exception as ex:  # pylint: disable=broad-except
+                    error = ex
+                    template_path = prepare_crash_report(
+                        error, filepath, self.crash_file_path
+                    )
+                if error is not None:
+                    msg = get_fatal_error_message(filepath, template_path)
+                    if isinstance(error, AstroidError):
+                        symbol = "astroid-error"
+                        msg = (filepath, msg)
+                    else:
+                        symbol = "fatal"
+                    self.add_message(symbol, args=msg)
```

Comment:
Why not do everything in the except handler? That way you wouldn't need to reassign the exception.